### PR TITLE
fix: install libreadline-dev on CI

### PR DESCRIPTION
**Pull request by luau-project**
## Description

Recently, GitHub updated their Ubuntu runners labeled as ```ubuntu-latest``` to run under Ubuntu 24.04.

> [!NOTE]
> 
> Previously, ```ubuntu-latest``` ran on Ubuntu 22.04.

Due this change, the package ```libreadline-dev``` is not installed anymore by default, causing the current (Ubuntu) CI to fail while building Lua even for unrelated changes (notably [https://github.com/lgi-devs/lgi/pull/331](https://github.com/lgi-devs/lgi/pull/331)), because ```readline``` is missing in the system.

## Changes

* Added a step to install ```libreadline-dev``` before the build of Lua on (Ubuntu) CI;
* Minor edits in the (Ubuntu) CI regarding the ```LUA_VERSION``` environment variable;
* Updated MSVC CI to use the latest LuaInstaller version to install Lua ([Release notes](https://github.com/luau-project/LuaInstaller/releases/latest)).

### CODE REVIEW

File: .github/workflows/msvc-ci.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
     
     env:
       WINGTK_URL: https://github.com/wingtk/gvsbuild/releases/download/2024.10.0/GTK${{ matrix.gtk-major-version }}_Gvsbuild_2024.10.0_x64.zip
-      LUAINSTALLER_URL: https://github.com/luau-project/LuaInstaller/releases/download/v0.3.0.0/LuaInstaller.Console-v0.3.0.0-x64.zip
+      LUAINSTALLER_URL: https://github.com/luau-project/LuaInstaller/releases/download/v0.5.0.0/LuaInstaller.Console-v0.5.0.0-x64.zip
```
Comment (psychon) [inline .github/workflows/msvc-ci.yml:29]: What's this? How does this bring in libreadline?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,18 +15,19 @@ jobs:
       matrix:
         version: [5.1.5, 5.2.4, 5.3.6, 5.4.7]
 
+    env:
+      LUA_VERSION: ${{ matrix.version }}
+
```
Comment (psychon) [inline .github/workflows/ci.yml:20]: Just to confirm: This is an unrelated refactoring that has no effect on libreadline availability, right?

File: .github/workflows/msvc-ci.yml (hunk 1 above)
Comment (luau-project) [reply .github/workflows/msvc-ci.yml:29]: this is unrelated to ubuntu CI. Just updating links.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (luau-project) [reply .github/workflows/ci.yml:20]: this is unrelated to ubuntu CI. Just updating links.

**Comment by luau-project** [comment]
This PR is on hold for now. On my personal fork just before opening the PR (with a small modification to CI run on any branch), I didn't get any issue (check [https://github.com/luau-project/lgi/actions/runs/12501449208](https://github.com/luau-project/lgi/actions/runs/12501449208)).

There seems to be some strange behavior going on Mesa VK, which I'm searching on the internet.

**Comment by luau-project** [comment]
## Update

TLDR;

the solution is to force software rendering on CI.

## In-depth explanation

> [!NOTE]
> 
> In my fork, even though the jobs were running on images defined as ```ubuntu-latest```, they were in fact ran on Ubuntu 22.04 (GitHub is in the transition process). This is the reason I didn't catch the error on my fork's CI.

Analyzing the upstream discussion at [https://gitlab.freedesktop.org/mesa/mesa/-/issues/10293](https://gitlab.freedesktop.org/mesa/mesa/-/issues/10293), and inspecting the diffs of related merges [https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/27836/diffs](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/27836/diffs), I found that forcing software rendering on Ubuntu 24.04 would solve the problem.

Thus, at top, the following excerpt forces software rendering.

```yaml
env:
  # Force software rendering on CI, because
  # we were getting a CI error on Ubuntu 24.04 as follows:
  #   MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER))
  # For more information, see https://github.com/lgi-devs/lgi/pull/332
  LIBGL_ALWAYS_SOFTWARE: 1
```

As you can see, it worked fine on Ubuntu 24.04 [https://github.com/lgi-devs/lgi/actions/runs/12502953910/job/34882613489#step:1:4](https://github.com/lgi-devs/lgi/actions/runs/12502953910/job/34882613489#step:1:4).

In my view, the PR is ready for review.

**Comment by psychon** [comment]
Thanks for investigating and fixing this!
